camel/storages/vectordb_storages/qdrant.py

```diff
@@ -0,0 +1,282 @@
+                        "is different from the embedding dim "
+                        f"({self.vector_dim}).")
+                return
+            except ValueError:
+                pass
+        self.collection = collection or datetime.now().isoformat()
+        self._create_collection(
+            collection=self.collection,
+            size=self.vector_dim,
+            distance=distance,
+        )
+        self.distance = distance
+        self.del_collection = del_collection
+
+    def __del__(self):
+        r"""Deletes the collection if :obj:`del_collection` is set to
+        :obj:`True`.
+        """
+        if self.del_collection:
+            self._delete_collection(self.collection)
+
+    def _create_collection(
+        self,
+        collection: str,
+        size: int,
+        distance: VectorDistance = VectorDistance.DOT,
+        **kwargs: Any,
+    ) -> None:
+        r"""Creates a new collection in the database.
+
+        Args:
+            collection (str): Name of the collection to be created.
+            size (int): Dimensionality of vectors to be stored in this
+                collection.
+            distance (VectorDistance, optional): The distance metric to be used
+                for vector similarity. (default: :obj:`VectorDistance.DOT`)
+            **kwargs (Any): Additional keyword arguments.
+        """
+        distance_map = {
+            VectorDistance.DOT: Distance.DOT,
+            VectorDistance.COSINE: Distance.COSINE,
+            VectorDistance.EUCLIDEAN: Distance.EUCLID,
+        }
+        self._client.recreate_collection(
+            collection_name=collection,
+            vectors_config=VectorParams(
+                size=size,
+                distance=distance_map[distance],
+            ),
+            **kwargs,
+        )
+
+    def _delete_collection(
+        self,
+        collection: str,
+        **kwargs: Any,
+    ) -> None:
+        r"""Deletes an existing collection from the database.
+
+        Args:
+            collection (str): Name of the collection to be deleted.
+            **kwargs (Any): Additional keyword arguments.
+        """
+        self._client.delete_collection(collection_name=collection, **kwargs)
+
+    def _check_collection(self, collection: str) -> Dict[str, Any]:
+        r"""Retrieves details of an existing collection.
+
+        Args:
+            collection (str): Name of the collection to be checked.
+
+        Returns:
+            Dict[str, Any]: A dictionary containing details about the
+                collection.
+        """
+        # TODO: check more information
+        collection_info = self._client.get_collection(
+            collection_name=collection)
+        vector_config = collection_info.config.params.vectors
+        return {
+            "vector_dim":
+            vector_config.size
+            if isinstance(vector_config, VectorParams) else None,
+            "vector_count":
+            collection_info.vectors_count,
```

Comment:
add more check types
```suggestion
            collection_info.vectors_count,
            "status":
            collection_info.status,
            "vectors_count":
            collection_info.vectors_count,
            "config":
            collection_info.config,
```